### PR TITLE
Feature 2 creating a listing

### PR DESCRIPTION
**Pull request by danielqiuu**
I have finished the code for creating a listing, as well as the listing class (that you guys can all access).

### CODE REVIEW

**Comment by MarkianV** [comment]
I noticed the Getters and the Setters for all the different attributes on Listing are not implemented. We agreed to implement getters and setters for all attributes. There's also no documentation for any of the methods (Javadoc atleast).

**Comment by liu-samuel** [comment]
look good

**Review by liu-samuel — APPROVED** [review]
looks good

**Review by claregillis — APPROVED** [review]
Looks good

**Review by helenastra — APPROVED** [review]
good

File: src/main/java/features/Listing.java [hunk 1]
```diff
@@ -1,4 +1,41 @@
 package features;
+import entities.User;
 
+import java.time.*; // to store date
 public class Listing {
+    String title;
+    int id; // String or int -> dependent on database implementation
+    LocalDate ld; // Represents a date (year, month, day (yyyy-MM-dd))
```
Comment (lukecheseldine) [inline src/main/java/features/Listing.java:8]: can we change this to date or dateAdded

File: src/main/java/features/Listing.java [hunk 2]
```diff
@@ -1,4 +1,41 @@
...
+    float price;
+    User seller;
+    String description;
+    String [] images;
```
Comment (lukecheseldine) [inline src/main/java/features/Listing.java:12]: is this a list of paths to the images? also, `String[] images;`

File: src/main/java/forms/CreateListingForm.java [hunk 3]
```diff
@@ -1,4 +1,42 @@
 package forms;
 
-public class CreateListingForm {
+import entities.User;
+import features.Listing;
+
+public class CreateListingForm extends Form{
```
Comment (lukecheseldine) [inline src/main/java/forms/CreateListingForm.java:6]:
```suggestion
public class CreateListingForm extends Form {
```

File: src/main/java/features/Listing.java [hunk 4]
```diff
@@ -1,4 +1,41 @@
 package features;
+import entities.User;
 
+import java.time.*; // to store date
 public class Listing {
+    String title;
```
Comment (lukecheseldine) [inline src/main/java/features/Listing.java:6]: should these be private attributes?

File: src/main/java/forms/CreateListingForm.java [hunk 5]
```diff
@@ -1,4 +1,42 @@
...
+        this.images = images;
+    }
+
+
+    public void submitListing(){
+        // This function will be called on the "submit" button click
+        // This will add a listing to the databse using the database interactor method.
+        String title = this.getTitle();
+        Listing L = new Listing(title, id,price, seller, description, images);
+        id++;
```
Comment (lukecheseldine) [inline src/main/java/forms/CreateListingForm.java:29]: did you mean for id to be a static attribute?

File: src/main/java/forms/Form.java [hunk 6]
```diff
@@ -13,7 +13,6 @@ public void setTitle (String title) {
         this.title = title;
     }
```
Comment (lukecheseldine) [inline src/main/java/forms/Form.java:15]: thanks for removing this!

File: src/main/java/forms/CreateListingForm.java [hunk 7]
```diff
@@ -1,4 +1,42 @@
...
+
+        // addListing(L);
+    }
+
+    @Override
+    protected boolean validateForm() {
+        return false;
```
Comment (lukecheseldine) [inline src/main/java/forms/CreateListingForm.java:36]: you can go ahead and add validation if you'd like. things like checking that the User actually exists, the price is positive, maybe a char limit on description or title...

**Review by lukecheseldine — CHANGES_REQUESTED** [review]
overall really good work daniel! this is a lot of code and a really good start! i just left some comments about minor things, should be quick fixes

File: src/main/java/forms/CreateListingForm.java (hunk 5 above)
Comment (danielqiuu) [reply src/main/java/forms/CreateListingForm.java:36]: Fixed

**Comment by danielqiuu** [comment]
> 

Fixed!

**Review by MarkianV — APPROVED** [review]
Seems good, I appreciate the documentation for the attributes of Listing, makes it easier to track.

**Review by lukecheseldine — APPROVED** [review]
looks good! remember to reformat code in intellij next time tho

**Review by liu-samuel — APPROVED** [review]
looks good to me!

**Pull request merged**